replica/database.cc

```diff
@@ -2346,18 +2346,24 @@ future<> database::flush_keyspace_on_all_shards(sharded<database>& sharded_db, s
     });
 }
 
-future<> database::snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush) {
+future<> database::snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush, bool snap_views) {
     if (!skip_flush) {
         co_await flush_table_on_all_shards(sharded_db, ks_name, table_name);
     }
     auto uuid = sharded_db.local().find_uuid(ks_name, table_name);
     auto table_shards = co_await get_table_on_all_shards(sharded_db, uuid);
     co_await table::snapshot_on_all_shards(sharded_db, table_shards, tag);
+    if (snap_views) {
+        for (const auto& vp : table_shards[this_shard_id()]->views()) {
+            bool snap_views_of_view = false;
+            co_await snapshot_table_on_all_shards(sharded_db, ks_name, vp->cf_name(), tag, skip_flush, snap_views_of_view);
```

Comment:
nit: `snap_views_of_view` is constant and can likely be dropped

Reply:
it's defined only to express the intention.
maybe should be defined as constexpr instead...